integrationTests/mock/transactionHandlerMock.go

```diff
@@ -55,6 +57,7 @@ func (thm *transactionHandlerMock) processSendTransaction(rw http.ResponseWriter
 
 	thm.mutTransactions.Lock()
 	thm.transactions[string(txHash)] = sendTxRequest
+	thm.notifier.notifyContract(sendTxRequest)
```

Comment:
should be moved outside of the mutex lock operation in order to avoid possible, future, deadlock bugs